src/main/java/org/cryptomator/ui/recoverykey/RecoveryKeyCreationController.java

```diff
@@ -1,52 +1,120 @@
 
 	private static final Logger LOG = LoggerFactory.getLogger(RecoveryKeyCreationController.class);
 
 	private final Stage window;
 	private final Lazy<Scene> successScene;
+	private final Lazy<Scene> recoverykeyExpertSettingsScene;
+	private final MasterkeyFileAccess masterkeyFileAccess;
 	private final Vault vault;
 	private final ExecutorService executor;
 	private final RecoveryKeyFactory recoveryKeyFactory;
 	private final StringProperty recoveryKeyProperty;
 	private final FxApplicationWindows appWindows;
 	public NiceSecurePasswordField passwordField;
+	private final IntegerProperty shorteningThreshold;
+	private final ObjectProperty<RecoveryActionType> recoverType;
+	private final ResourceBundle resourceBundle;
+	public FormattedLabel descriptionLabel;
+	public Button cancelButton;
+	public Button nextButton;
+	private final VaultListManager vaultListManager;
+	private final Dialogs dialogs;
+	private final Stage owner;
 
 	@Inject
-	public RecoveryKeyCreationController(@RecoveryKeyWindow Stage window, @FxmlScene(FxmlFile.RECOVERYKEY_SUCCESS) Lazy<Scene> successScene, @RecoveryKeyWindow Vault vault, RecoveryKeyFactory recoveryKeyFactory, ExecutorService executor, @RecoveryKeyWindow StringProperty recoveryKey, FxApplicationWindows appWindows, ResourceBundle resourceBundle) {
+	public RecoveryKeyCreationController(FxApplicationWindows appWindows, //
+										 @RecoveryKeyWindow Stage window, //
+										 @Named("keyRecoveryOwner") Stage owner, //
+										 @FxmlScene(FxmlFile.RECOVERYKEY_SUCCESS) Lazy<Scene> successScene, //
+										 @FxmlScene(FxmlFile.RECOVERYKEY_EXPERT_SETTINGS) Lazy<Scene> recoverykeyExpertSettingsScene, //
+										 @RecoveryKeyWindow Vault vault, //
+										 RecoveryKeyFactory recoveryKeyFactory, //
+										 MasterkeyFileAccess masterkeyFileAccess, //
+										 ExecutorService executor, //
+										 @RecoveryKeyWindow StringProperty recoveryKey, //
+										 @Named("shorteningThreshold") IntegerProperty shorteningThreshold, //
+										 @Named("recoverType") ObjectProperty<RecoveryActionType> recoverType, //
+										 VaultListManager vaultListManager, //
+										 ResourceBundle resourceBundle, //
+										 Dialogs dialogs) {
 		this.window = window;
-		window.setTitle(resourceBundle.getString("recoveryKey.display.title"));
 		this.successScene = successScene;
+		this.recoverykeyExpertSettingsScene = recoverykeyExpertSettingsScene;
 		this.vault = vault;
 		this.executor = executor;
 		this.recoveryKeyFactory = recoveryKeyFactory;
 		this.recoveryKeyProperty = recoveryKey;
 		this.appWindows = appWindows;
+		this.recoverType = recoverType;
+		this.resourceBundle = resourceBundle;
+		this.masterkeyFileAccess = masterkeyFileAccess;
+		this.shorteningThreshold = shorteningThreshold;
+		this.vaultListManager = vaultListManager;
+		this.owner = owner;
+		this.dialogs = dialogs;
+	}
+
+	@FXML
+	public void initialize() {
+		if (recoverType.get().equals(RecoveryActionType.SHOW_KEY)) {
```

Comment:
You can compare enums directly with `==` (see for example https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals). Makes it also easier to read.